Core/Object Arts/Dolphin/Base/BlockClosure.cls

```diff
@@ -549,6 +549,33 @@ stackTempCount: anInteger
 	info := (info bitAnd: ##((16rFF << 15) bitInvert)) 
 				bitOr: ((anInteger bitAnd: 16rFF) bitShift: 15)!
 
+storeOn: aStream
+	"Append, to aStream, a <String> whose characters can recreate the receiver."
+
+	| m block class selector string |
+	m := self method.
+	(m notNil and: [m isExpression]) ifTrue: [	"We can print the source"
```

Comment:
Source form could also be used for any completely clean block (any that are non-full and non-copying, and don't need self), as these can be taken out of the method context. I'm not sure how easy it is to find the source range, however, at least not without using the parser.